src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -46,62 +55,73 @@ public Command setState(RobotState state) {
 	//@formatter:off
 	public Command idle() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE)
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+				funnelStateHandler.setState(FunnelState.STOP)
 		);
 	}
 
 	public Command intake() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.NOTE))
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.NOTE)),
+				funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
 		);
 	}
 
 	public Command preSpeaker() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER))
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER)),
+				funnelStateHandler.setState(FunnelState.STOP)
 		);
 	}
 
 	public Command speaker() {
 		return new ParallelCommandGroup(
+				funnelStateHandler.setState(FunnelState.SPEAKER),
 				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER))
 		);
 	}
 
 	public Command preAmp() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP))
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)),
+				funnelStateHandler.setState(FunnelState.STOP)
 		);
 	}
 
 	public Command amp() {
 		return new ParallelCommandGroup(
 				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP))
+						.until(() -> swerve.isAtHeading(Field.getAngleToAmp())),
```

Comment:
Should one tab more to the left
```suggestion
					.until(() -> swerve.isAtHeading(Field.getAngleToAmp())),
```